Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -614,64 +681,32 @@ bool UserCmd_Process(uint client, const wstring &wscCmd)
 	return false;
 }
 
-void __stdcall JumpInComplete(uint iSystemID, uint iShip)
-{
-	returncode = DEFAULT_RETURNCODE;
-	const uint iClientID = HkGetClientIDByShip(iShip);
-	
-	// After completing a jump, if it's in our dockedships map, we update the current system
-	if(mobiledockClients.find(iClientID) != mobiledockClients.end())
-	{
-		pub::SpaceObj::GetSystem(iShip, mobiledockClients[iClientID].carrierSystem);
-	}
-	
-}
-
-void __stdcall CharacterSelect(struct CHARACTER_ID const & cId, unsigned int iClientID)
-{
-
-	returncode = DEFAULT_RETURNCODE;
-	
-	// Attempt to load the data for this ship if it exists. Trim off the .fl at the end of the file name.
-	string charFilename = cId.szCharFilename;
-	charFilename = charFilename.substr(0, charFilename.size() - 3);
-
-	LoadShip(charFilename);
-
-}
-
 void __stdcall DisConnect(uint iClientID, enum EFLConnection p2)
 {
-
 	returncode = DEFAULT_RETURNCODE;
-	// Is the disconnecting user a part of the docking module plugin at the moment?
-	if(mobiledockClients.find(iClientID) != mobiledockClients.end())
-	{
-		// Is this a carrier?
-		if(!mobiledockClients[iClientID].mapDockedShips.empty() || mobiledockClients[iClientID].iDockingModulesAvailable > 0)
-		{
-			wstring shipFileName;
-			wstring charName = reinterpret_cast<const wchar_t*>(Players.GetActiveCharacterName(iClientID));
 
-			HkGetCharFileName(charName, shipFileName);
-			SaveDockInfoCarrier(shipFileName ,iClientID, mobiledockClients[iClientID]);
-		}
+	// If the ship was docked to someone, erase it from docked ship list.
+	if (!mobiledockClients[iClientID].wscDockedWithCharname.empty())
+	{
+		uint carrierClientID = HkGetClientIdFromCharname(mobiledockClients[iClientID].wscDockedWithCharname);
 
-		// Is this a carried ship?
-		else if(!empty(mobiledockClients[iClientID].wscDockedWithCharname))
+		// If carrier is present at server - do it, if not - whatever. Plugin erases all associated client data after disconnect. 
+		if (carrierClientID != -1)
 		{
-			wstring shipFileName;
-			wstring charName = reinterpret_cast<const wchar_t*>(Players.GetActiveCharacterName(iClientID));
-
-			HkGetCharFileName(charName, shipFileName);
-			SaveDockInfoCarried(shipFileName, iClientID, mobiledockClients[iClientID]);
+			wstring charname = (const wchar_t*)Players.GetActiveCharacterName(iClientID);
+			mobiledockClients[carrierClientID].mapDockedShips.erase(charname);
 		}
 
-		mobiledockClients.erase(iClientID);
+		// Redirect client to last real base.

```

Comment:
what if they last docked on a POB?

Reply:
They will be redirected to proxy base in same system with the POB.